src/server/Ebenezer/EbenezerDlg.cpp

```diff
@@ -2613,29 +2613,41 @@ BOOL CEbenezerDlg::PreTranslateMessage(MSG * pMsg) {
     return CDialog::PreTranslateMessage(pMsg);
 }
 
-BOOL CEbenezerDlg::LoadNoticeData() {
-    CString    ProgPath = GetProgPath();
-    CString    NoticePath = ProgPath + "Notice.txt";
-    CString    buff;
-    CStdioFile txt_file;
-    int        count = 0;
-
-    if (!txt_file.Open(NoticePath, CFile::modeRead)) {
-        AfxMessageBox("cannot open Notice.txt!!");
-        return FALSE;
-    }
+bool CEbenezerDlg::LoadNoticeData() {
+    memset(&m_ppNotice, 0, sizeof(m_ppNotice));
+    if (!fs::exists("Notice.txt")) {
+        // Create an empty file, so that people know of its existence
+        std::ofstream fileEmpty("Notice.txt");
+        return true;
+    }
+
+    std::ifstream inFile("Notice.txt");
+    if (!inFile) {
+        AfxMessageBox("ERROR: Notice.txt could not be opened.\n");
+        return false;
+    }
```

Comment:
You probably already know that `AfxMessageBox` is WinAPI specific, but either way, eventually I think it will be good changing the server files to be console based if you want to make it more portable. Another alternative is using some library, such as Qt or a quick solution: https://github.com/samhocevar/portable-file-dialogs

Reply:
Yes, this is definitely on the radar. Nice! I didn't know about `portable-file-dialogs`. I'm even thinking of making the server start either in GUI or Console mode that is configurable from the settings file. As for GUI, we could also use imgui.

Thanks for the feedback!